src/cairo_runner.rs

```diff
@@ -173,6 +175,16 @@ impl PyCairoRunner {
             .map(PyExecutionResources)
             .map_err(to_py_error)
     }
+
+    pub fn get_segment_used_size(&self, index: usize, py: Python) -> PyResult<PyObject> {
+        Ok(self
+            .pyvm
+            .vm
+            .borrow()
+            .get_segment_used_size(index)
+            .ok_or_else(|| PyTypeError::new_err(MEMORY_GET_SEGMENT_USED_SIZE_MSG))?
```

Comment:
I'm not sure if PyTypeError is the correct Error type to use here. But aside from that LGTM